view.php

```diff
@@ -28,13 +30,32 @@
 $PAGE->set_title($record->name);
 $PAGE->set_heading($record->name);
 
+if ($insert) {
+    $DB->insert_record('cpdlogbook_entries', [
+            'user' => $USER->id,
+            'cpdlogbook' => $record->id,
+            'name' => 'test',
+            'time' => time(),
+    ]);
+    redirect($PAGE->url);
+}
+
 echo $OUTPUT->header();
 
 echo html_writer::alist([
         'id' => $record->id,
         'course' => $record->course,
         'name' => $record->name,
         'totalpoints' => $record->totalpoints,
+        'info' => $record->intro,
+        'introformat' => $record->introformat,
 ]);
 
+echo html_writer::link($PAGE->url.'&insert=true', 'Insert a record');
+
+$table = new table_sql('cpdlogbook_id');
+$table->set_sql('*', '{cpdlogbook_entries}', 'cpdlogbook='.$record->id.' AND user='.$USER->id);
```

Comment:
This is a great start, A little thing: You don't want to inline the sql params like this you want to send them as params

$table->set_sql($fields, $from, $where, $params);

something roughly like:

$table->set_sql('*', '{cpdlogbook_entries}', 'cpdlogbook=? AND user=?', [$record->id, $USER->id] );

This is how sql injection vulnerabilities sneak in otherwise